### DISCUSSION TITLE
Contours with slope lines - how did you fix that?

### DISCUSSION THREAD

**Discussion by stefankinell** [manudona92/MapantES#72] [Q&A]
Hi,
I see that you have managed to display contours with slope lines in a very nice way on mapant.es. On my kartor.gokartor.se I have not found a good way to fix this and therefore use the older way of using purple color for depressions. Can you share how you did managed to do this? I know you use a different method with the server in the background but I believe you are still using the same methodology to render the contours with pullauta.

Thanks
Stefan

**Comment by agustincaballero** [comment]
Hi, Stefan. We work with a postgis database in the background, so we have every element in the map stored as vector.

What we do is that we process the out2.dxf file generated by Karttapullautin, that file contains all regular contour lines and also depressions, labelled with "depression", "depression_intermed" and "depression_index".

We then store these in the server with special (and invented) isom codes 101.90, 102.90 and 103.90.

The map is rendered by a mapserver, in which it is quite easy to tell it to render this isom codes as a line with tags (I can give you details on exactly how to do this if you are interested).

**Reply by stefankinell** [reply]
I think I have been locked into the assumption that it would not be possible to interpret automatically in what direction the slop lines would point - risking it would point the wrong direction. But now when I did a test in one of my OCAD-files all the depressions rendered with the slope line in the right direction. Interesting. I think I have to consider changing the purple depressions to proper brown in my next version of the Swedish map.

Have you had any occasions where the slope lines are in the wrong direction so to say?

**Reply by agustincaballero** [reply]
No, we haven't found any case like that, the line is always generated so that the lower part is in the right (meaning not left) part of the line.

For example: 
https://mapant.es/#x=-457073.99&y=5276677.57&resolution=1.4142&layers=vector
https://mapant.es/#x=-505708.67&y=4434631.67&resolution=0.5&layers=vector

**Reply by stefankinell** [reply]
Brilliant, then I learned something new today. I do not know why I decided it would not work without trying. Thanks for helping me realize I was wrong :-D
